pkg/kubefedctl/federate/federate.go

```diff
@@ -83,17 +84,35 @@ type federateResource struct {
 	outputYAML        bool
 	enableType        bool
 	federateContents  bool
+	filename          string
 }
 
 func (j *federateResource) Bind(flags *pflag.FlagSet) {
 	flags.StringVarP(&j.resourceNamespace, "namespace", "n", "default", "The namespace of the resource to federate.")
 	flags.StringVarP(&j.output, "output", "o", "", "If provided, the resource that would be created in the API by the command is instead output to stdout in the provided format.  Valid format is ['yaml'].")
 	flags.BoolVarP(&j.enableType, "enable-type", "e", false, "If true, attempt to enable federation of the API type of the resource before creating the federated resource.")
 	flags.BoolVarP(&j.federateContents, "contents", "c", false, "Applicable only to namespaces. If provided, the command will federate all resources within the namespace after federating the namespace.")
+	flags.StringVarP(&j.filename, "filename", "f", "", "If specified, the provided yaml file will be used as the input for target resources to federate. Only --output will be accepted from command line with this option. Other flag options if provided will be ignored")
 }
 
 // Complete ensures that options are valid.
 func (j *federateResource) Complete(args []string) error {
+	if j.output == "yaml" {
+		j.outputYAML = true
+	} else if len(j.output) > 0 {
+		return errors.Errorf("Invalid value for --output: %s", j.output)
+	}
+
+	if len(j.filename) > 0 {
+		if len(args) > 0 {
+			return errors.Errorf("Invalid args: %v, option '--filename' does not take any args", args)
```

Comment:
As per a conversation we've already had on previous PRs, embedding arbitrary user input in a message intended to indicate an error risks obscuring the error.  Please reword so that the args are appended to the error message.